app/controllers/course/registrations_controller.rb

```diff
@@ -0,0 +1,131 @@
+class Course::RegistrationsController < Course::ModuleController
+  load_and_authorize_resource through: :course
+  before_action :ensure_course_open, :ensure_not_registered, :update_if_exist, only: :create
+
+  def index #:nodoc:
+    @staff_requests = @registrations.staff
+    @student_requests = @registrations.student
+  end
+
+  def create #:nodoc:
+    @registration.user = current_user
+    @registration.save!
+    @registration.update_attributes(registration_params)
+
+    # TODO: Notify lecturer
+    redirect_to course_path(@course), flash: { notice: t('.request_format') %
```

Comment:
Can I use save! and handle exceptions with a 
`rescue_from ActiveRecord::RecordInvalid, with: validation_failed`
This will be helpful when marketplace is added. When a course is full, the save will fail with the error message prompting the user to buy more seats for his course. The validation_failed call back will set the correct flash error message.